src/gp_engine.hpp

```diff
@@ -67,13 +78,15 @@ class GPEngine {
  private:
   static constexpr double min_branch_length_ = 1e-6;
   static constexpr double max_branch_length_ = 3.;
+  static constexpr double min_log_branch_length_ = -13.9;
+  static constexpr double max_log_branch_length_ = 1.1;
 
   int significant_digits_for_optimization_ = 6;
   double relative_tolerance_for_optimization_ = 1e-2;
   double step_size_for_optimization_ = 5e-4;
   size_t max_iter_for_optimization_ = 1000;
 
-  double log_marginal_likelihood_ = DOUBLE_NEG_INF;
+  EigenVectorXd log_marginal_likelihood_;
```

Comment:
Please describe the entries of this vector in a comment.

Reply:
Done.